src/main/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/Message.java

```diff
@@ -0,0 +1,70 @@
+package uk.nhs.digital.nhsconnect.lab.results.model.edifact;
+
+import lombok.Getter;
+import lombok.Setter;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class Message extends Section {
+
+    private static final String DEFAULT_GP_CODE = "9999";
+
+    @Getter(lazy = true)
+    private final MessageHeader messageHeader =
+        MessageHeader.fromString(extractSegment(MessageHeader.KEY));
+
+    /*
```

Comment:
I also don't see any benefit in committing the commented out code in this class. Again, we can add it later, if and when we need it :-)